tests/integration/test-repository.mjs

```diff
@@ -0,0 +1,167 @@
+/**
+ * Utilities for creating and managing test Git repositories
+ * These utilities help set up realistic test repositories for integration testing
+ */
+
+import fs from 'fs';
+import path from 'path';
+import os from 'os';
+import { execSync } from 'child_process';
+
+/**
+ * Creates a temporary Git repository for testing
+ * 
+ * @param {Object} options Configuration options
+ * @param {boolean} options.withDevelop Whether to include a develop branch
+ * @param {boolean} options.withFeature Whether to include a feature branch
+ * @param {boolean} options.withRemote Whether to set up a remote repository simulation
+ * @returns {Object} Repository information including paths and cleanup function
+ */
+export function createTestRepository(options = {}) {
+  const {
+    withDevelop = true,
+    withFeature = false,
+    withRemote = false
+  } = options;
+  
+  // Create a temporary directory
+  const testDir = fs.mkdtempSync(path.join(os.tmpdir(), 'git-manager-test-'));
+  
+  // Initialize git repository
+  execSync('git init', { cwd: testDir });
+  
+  // Configure git identity for the test repository
+  execSync('git config user.name "Test User"', { cwd: testDir });
+  execSync('git config user.email "test@example.com"', { cwd: testDir });
+  
+  // Create initial commit on main branch
+  fs.writeFileSync(path.join(testDir, 'README.md'), '# Test Repository\n');
+  execSync('git add README.md', { cwd: testDir });
+  execSync('git commit -m "Initial commit"', { cwd: testDir });
+  
+  // Ensure we're using main branch (newer git defaults to main, older to master)
+  try {
+    execSync('git branch -m master main', { cwd: testDir, stdio: 'ignore' });
+  } catch (e) {
+    // Branch already named main, ignore error
+  }
+  
+  // Create develop branch if requested
+  if (withDevelop) {
+    execSync('git checkout -b develop', { cwd: testDir });
```

Comment:
I would want you to use `api.js` to implement these commands so we test the codebase while we do the integration tests